File: python/FibonacciNumbers.py
import time

def naiveSol(n):
	if n <= 1:
		return n
	else:
		return naiveSol(n-1) + naiveSol(n-2)

def fastSol(n):
    arr = []
    arr.append(0)
    arr.append(1)
    
    for i in range(2, n + 1):
    	arr.append( arr[i - 1] + arr[i - 2] )

    return arr[n]

#n = int(input())

result = 1
n = 0
while result and n <= 45:
	#startTime = time.time()
	a = naiveSol(n)
	b = fastSol(n)
	if a == b:
		print(n, " - ok")
		n += 1
	else:
		printprint(n, " - fail")
		result = 0
	#print("elapsed time: ", time.time() - startTime)

File: java/src/FibonacciNumbers.java
import java.io.*;
import java.util.*;

public class FibonacciNumbers {
	
	public static long naiveSol(int n) {
		if (n <= 1) {
			return n;
		}
		
		return naiveSol(n - 1) + naiveSol(n - 2);
	}
	
	public static long fastSol(int n) {
		long[] arr = new long[n + 1];
		arr[0] = 0;

		if (n > 0) {
			arr[1] = 1;
		}

		for(int i = 2 ; i < n + 1 ; i++) {
			arr[i] = arr[i - 1] + arr[i - 2];
		}
		
		return arr[n];
	}

	public static void main(String[] args) {
		/*FastScanner scanner = new FastScanner(System.in);
		int n = scanner.nextInt();
		System.out.println(naiveSol(n));
		System.out.println(fastSol(n));*/
		
		boolean result = true;
		int n = 0;
		while (result && n <= 45) {
			if (naiveSol(n) == fastSol(n)) {
				System.out.println(n + " -  ok" );
				n++;
			} else {
				System.out.println(n + " -  fail" );
				result = false;
			}
		}
	}
	
	static class FastScanner {
		BufferedReader br;
		StringTokenizer st;

		public FastScanner(InputStream is) {
			try {
				br = new BufferedReader(new InputStreamReader(is));
			} catch (Exception e) {
				e.printStackTrace();
			}
		}
		
		public String next() {
			while (st == null || !st.hasMoreTokens()) {
				try {
					st = new StringTokenizer(br.readLine());
				} catch (Exception e) {
					e.printStackTrace();
				}
			}
			return st.nextToken();
		}
		
		public int nextInt() {
			return Integer.parseInt(next());
		}
	}
}
